### PR TITLE
NhsE IG/Program validation - option to ignore warnings

### PR DESCRIPTION
**Pull request by APageNHS**

### CODE REVIEW

File: src/common.js.ts [hunk 1]
```diff
@@ -252,89 +252,96 @@ function getErrorOrWarningFull(issue: OperationOutcomeIssue) {
...
-            if (issue.diagnostics.includes('https://fhir.hl7.org.uk/ValueSet/UKCore-MedicationCode')) return false
-            if (issue.diagnostics.includes('https://fhir.hl7.org.uk/ValueSet/UKCore-VaccineCode')) return false
-        }
-*/
-        if (issue.diagnostics.includes('Unknown code')) {
-            if (issue.diagnostics.includes('https://fhir.nhs.uk/CodeSystem/NHSDigital-SDS-JobRoleCode')) return false
-        }
+        
         if (issue.diagnostics.includes('None of the codings provided are in the value set')) {
             if (issue.diagnostics.includes('https://fhir.nhs.uk/CodeSystem/NHSDigital-SDS-JobRoleCode')) return false
```
Comment (ryma2fhir) [inline src/common.js.ts:300]: There is no longer a codesystem named NHSDigital-SDS-JobRoleCode

**Review by ryma2fhir — APPROVED** [review]

File: src/common.js.ts [hunk 2]
```diff
@@ -252,89 +252,96 @@ function getErrorOrWarningFull(issue: OperationOutcomeIssue) {
...
+            // languages, known issue!
+            if (issue.diagnostics.includes('Inappropriate CodeSystem URL') && issue.diagnostics.includes('for ValueSet: http://hl7.org/fhir/ValueSet/all-languages')) {
+                return false
+            }
+            
             // Ignore LOINC Errors for now
             if (issue.diagnostics.includes('http://loinc.org')) return false;
+            
+            // ignore dm+d / read errors
+            if (issue.diagnostics.includes('Code system https://dmd.nhs.uk/ could not be resolved.')) return false
```
Comment (ryma2fhir) [inline src/common.js.ts:327]: dm+d is in ontoserver sho should resolve. I commented this line and line 269 out on my fork and no additional errors showed in the log https://github.com/ryma2fhir/RYAN-FHIR-R4-UKCORE-STAGING-MAIN/actions/runs/7422530696

File: src/common.js.ts [hunk 3]
```diff
@@ -252,89 +252,96 @@ function getErrorOrWarningFull(issue: OperationOutcomeIssue) {
...
+            
+            // there Errors need checking
+            /*
+            if (issue.diagnostics.includes('could not be resolved, so has not been checked')) return false;
+
             // fault with current 5.5.1 validation
             if (issue.diagnostics.includes('http://hl7.org/fhir/ValueSet/units-of-time')) return false;
             if (issue.diagnostics.includes('NHSNumberVerificationStatus')) return false;
             if (issue.diagnostics.includes('Validation failed for \'http://example.org/fhir')) return false;
             if (issue.diagnostics.includes('Unrecognised property \'@fhir_comments')) return false;
```
[the remark this answers is not in the document]
Comment (APageNHS) [reply src/common.js.ts]: removed all the old exceptions that were commented out

File: src/common.js.ts (hunk 1 above)
Comment (APageNHS) [reply src/common.js.ts]: Agreed, But there IS a fixed value slice in the NHSDigital PractitionerRole profile, keeping this check in

File: src/common.js.ts [hunk 4]
```diff
@@ -252,89 +252,96 @@ function getErrorOrWarningFull(issue: OperationOutcomeIssue) {
...
+        //THESE WARNINGS SHOULD ALWAYS ERROR
         if (issue.diagnostics.includes('incorrect type for element')) {
             return true;
         }
         if (issue.diagnostics.includes('Error HTTP 401')) {
             return true;
         }
-        if (issue.diagnostics.includes('Could not confirm that the codes provided are in the value set')) {
-            if (issue.diagnostics.includes('http://hl7.org/fhir/ValueSet/usage-context-type')) return false;
+        if (issue.diagnostics.includes('must be of the format')) {
```
[the remark this answers is not in the document]
Comment (APageNHS) [reply src/common.js.ts:263]: guessing it was possibly for this:

https://github.com/NHSDigital/IOPS-FHIR-Testing/blob/b4b4388aa5fb920f43a2f586a6e3b9d7395c128e/src/app/main/validate-fhir/validate-fhir.component.ts#L100

[the remark this answers is not in the document]
Comment (APageNHS) [reply src/common.js.ts:263]: will remove the check and see what happens

**Review by ryma2fhir — APPROVED** [review]

**Review by ryma2fhir — APPROVED** [review]

**Pull request merged**